src/components/Follow/FollowButton.tsx

```diff
@@ -0,0 +1,85 @@
+import { useCallback, useMemo } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+import useFollowUser from './mutations/useFollowUser';
+import useUnfollowUser from './mutations/useUnfollowUser';
+import Tooltip, { StyledTooltipParent } from 'components/Tooltip/Tooltip';
+import IconButton from 'components/IconButton/IconButton';
+import { useToastActions } from 'contexts/toast/ToastContext';
+
+type Props = {
+  userRef: any;
+  isFollowing: boolean;
+  loggedInUserId?: string;
+};
+
+export default function FollowButton({ userRef, isFollowing, loggedInUserId }: Props) {
+  const user = useFragment(
+    graphql`
+      fragment FollowButtonFragment on GalleryUser {
+        id
+        dbid
+        username
+        followers {
+          id
+        }
+        following {
+          id
+        }
+      }
+    `,
+    userRef
+  );
+
+  const followUser = useFollowUser();
+  const unfollowUser = useUnfollowUser();
+  const { pushToast } = useToastActions();
+
+  const handleFollowClick = useCallback(() => {
+    const optimisticNewFollowersList = [{ id: loggedInUserId }, ...user.followers];
+    followUser(user.dbid, optimisticNewFollowersList, user.following).then(() =>
+      pushToast({ message: `You have followed ${user.username}.` })
+    );
```

Comment:
nit: i think we can play around with `optimisticUpdater` so that we don't have to pass in the optimistic new followers list, but rather handle the append directly on the mutation: https://relay.dev/docs/guided-tour/updating-data/graphql-mutations/#optimistic-updater

so we'd use something like the following, and access our `store` to get the current followers, etc. i'll ask terence about what's ideal!

<img width="708" alt="image" src="https://user-images.githubusercontent.com/12162433/170084838-10a81d4a-3909-4de7-b2f1-36d3e5932fda.png">